phaser/builtin_steps.py

```diff
@@ -70,3 +76,73 @@ def filter_rows_step(batch, context, **kwargs):
                                     message=f"{num_dropped} rows dropped in filter_rows with '{func.__name__}'")
         return new_batch
     return filter_rows_step
+
+
+@row_step
+def flatten_all(row, context, **kwargs):
+    new_row = row.copy()
+    still_flattening = True
+    while still_flattening:
+        # Each iteration will pull nested data up one level. This will repeat until all nested data flattened.
+        still_flattening = False
+        for key, value in new_row.copy().items():
+            if isinstance(value, dict):
+                still_flattening = True
+                new_row, new_columns = _merge_values_if_no_collisions(new_row, key)
+    return new_row
+
+
+def _merge_values_if_no_collisions(row, key_name):
+    new_row = row.copy()
+    new_columns = []
+    value = row[key_name]
+    for inner_key, inner_value in value.items():
+        new_column_name = f"{key_name}__{inner_key}"
+        if new_column_name in row.keys():
+            raise DataErrorException(f"Error flattening nested data; key {key_name} already in row")
+        new_row[new_column_name] = inner_value
+        new_columns.append(new_column_name)
+    del new_row[key_name]
+    return new_row, new_columns
+
+
+def flatten_dict(column_name, keep_going=True):
```

Comment:
Is `keep_going` meant to be a signal to flatten the value down as many levels deep as there are dicts to flatten?  I think that is what the code is doing, but I'm not 100% sure.  If so, maybe `recurse` is "better" as a name?  There may also be a better name, or the problem is solved with a good ol' fashioned doc string.